homework/tests/homework.spec.js

```diff
@@ -0,0 +1,172 @@
+// @ts-nocheck
+const { test, expect } = require('@playwright/test');
+
+const data = [
+  'Prototype',
+  '1',
+  '2',
+  '3',
+  '4',
+  '5',
+  '6',
+  '7',
+  '8',
+  '9'
+]
+
+//Concatenate
+
+data.forEach(version => {
+  test.describe(version + ': Concatenate', () => {
+    test('Concatenating 2 and 3 results in 23', async ({ page }) => {
+      await page.goto('https://testsheepnz.github.io/BasicCalculator');
+      await page.selectOption('#selectBuild', { label: version});
+      await page.locator('#number1Field').type('2');
+      await page.locator('#number2Field').type('3');
+      await page.selectOption('#selectOperationDropdown', {label: 'Concatenate'});
+      await page.locator('#calculateButton').click();
+  
+      await expect(page.locator('#numberAnswerField')).toHaveValue('23');
+      await page.pause();
+    });
+  });
+});
+
+//Devide by zero error message
+
+data.forEach(version => {
+  test.describe(version + ': Devide by 0', () => {
+    test('Deviding number by 0 getting 0', async ({ page }) => {
+      await page.goto('https://testsheepnz.github.io/BasicCalculator');
+      await page.selectOption('#selectBuild', { label: version});
+      await page.locator('#number1Field').type('2');
+      await page.locator('#number2Field').type('0');
+      await page.selectOption('#selectOperationDropdown', {label: 'Divide'});
+      await page.locator('#calculateButton').click();
+  
+      await expect(page.locator('#numberAnswerField')).toHaveValue('');
+      await page.pause();
+    });
+  });
+});
+
+//If ts number?
+
+data.forEach(version => {
+  test.describe(version + ': Is it number', () => {
+    test('Check if letter "a" is a number result is not a number', async ({ page }) => {
```

Comment:
I literally do not understand what are you testing here based on this TC name 😄  Could you try changing it somehow?

Reply:
And if you want to assert whether something is a number - you can use JS object class:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger

Reply:
There are versions that able to add sybols and letter. So this test had to search for any given anwser in result field. There should be no result in this case.